src/main/scala/com/databricks/labs/overwatch/pipeline/Initializer.scala

```diff
@@ -275,25 +275,33 @@ class Initializer(config: Config) extends SparkSessionWrapper {
-    // TODO - Validate if token has access to necessary assets. Warn/Fail if not
-    if (tokenSecret.nonEmpty && !disableValidations && !config.isLocalTesting) {
-      if (tokenSecret.get.scope.isEmpty || tokenSecret.get.key.isEmpty) {
-        throw new BadConfigException(s"Secret AND Key must be provided together or neither of them. " +
-          s"Either supply both or neither.")
-      }
-      val scopeCheck = dbutils.secrets.listScopes().map(_.getName()).toArray.filter(_ == tokenSecret.get.scope)
-      if (scopeCheck.length == 0) throw new BadConfigException(s"Scope ${tokenSecret.get.scope} does not exist " +
-        s"in this workspace. Please provide a scope available and accessible to this account.")
-      val scopeName = scopeCheck.head
-
-      val keyCheck = dbutils.secrets.list(scopeName).toArray.filter(_.key == tokenSecret.get.key)
-      if (keyCheck.length == 0) throw new BadConfigException(s"Key ${tokenSecret.get.key} does not exist " +
-        s"within the provided scope: ${tokenSecret.get.scope}. Please provide a scope and key " +
-        s"available and accessible to this account.")
-
-      config.registerWorkspaceMeta(Some(TokenSecret(scopeName, keyCheck.head.key)))
-    } else config.registerWorkspaceMeta(None)
+
+    rawParams.tokenSecret.map {
+        case databricksSecret: TokenSecret =>
+          val apiToken = SecretTools(databricksSecret).getApiToken
+          // validate token secret requirements
+          // TODO - Validate if databricks token has access to necessary assets. Warn/Fail if not
+          if (apiToken.nonEmpty && !disableValidations && !config.isLocalTesting) {
+            if (databricksSecret.scope.isEmpty || databricksSecret.key.isEmpty) {
+              throw new BadConfigException(s"Secret AND Key must be provided together or neither of them. " +
+                s"Either supply both or neither.")
+            }
+            val scopeCheck = dbutils.secrets.listScopes().map(_.getName()).toArray.filter(_ == databricksSecret.scope)
+            if (scopeCheck.length == 0) throw new BadConfigException(s"Scope ${databricksSecret.scope} does not exist " +
+              s"in this workspace. Please provide a scope available and accessible to this account.")
+            val scopeName = scopeCheck.head
+
+            val keyCheck = dbutils.secrets.list(scopeName).toArray.filter(_.key == databricksSecret.key)
+            if (keyCheck.length == 0) throw new BadConfigException(s"Key ${databricksSecret.key} does not exist " +
+              s"within the provided scope: ${databricksSecret.scope}. Please provide a scope and key " +
+              s"available and accessible to this account.")
+
+            config.registerWorkspaceMeta(Some(TokenSecret(scopeName, keyCheck.head.key)))
+          } else config.registerWorkspaceMeta(None)
+
+        case awsSecret: AwsTokenSecret => config.registerWorkspaceMeta(Some(awsSecret))
+        case _ => config.registerWorkspaceMeta(None)
```

Comment:
Looks like here we change previous behavior.

In the master branch we apply 
```scala
config.registerWorkspaceMeta(None)
```
if rawParams.tokenSecret is None. 

Now we apply it only if awParams.tokenSecret is Some but not in (TokenSecret, AwsTokenSecret). Is this correct behaviour?

Reply:
To keep previous behaviour (registerWorkspaceMeta(None))  in case we passed None or it's local testing or we want to disable all the validations we need to move this check up before mapping over structure passed...Fixed